### DISCUSSION TITLE
Device locations float and move, cannot stop

### DISCUSSION THREAD

**Discussion by warriorsoul15** [mattieserver/netbox-topology-views#203] [Q&A] [answered]
I cannot seem to be able to drag a device to a place on the screen and have it stay. it snaps back and floats.  I am using Chrome (latest).  Is this a browser issue or plugin issue

**Comment by bharath-ravindranath** [comment]
There is a configuration `allow_coordinates_saving` that you should enable (checkout the documentation). Once done (and `netbox` is restarted), you will see an option in topology view to `Save Coordinates` (a checkbox). 

FWIW, even after you do this the objects may float (it did for me). You may want to drag each item once so that the co-ordinates are updated (just drag them anywhere). Then refresh the page. After this, you should see all object stay in place. You can drag them to their intended places now. Hope it helps

This might be a duplicate of #25

**Comment by phlpr** [comment]
Hi,

we have the same problem with the following versions:

NB-Version: 4.0.10
Plugin-Version: 4.0.1

We also tested NB-Version 4.0.6 and Plugin-Version 4.0.0.

The plugin is configured with both parameters enabled:
```
PLUGINS_CONFIG = {
    'netbox_topology_views': {
        'static_image_directory': 'netbox_topology_views/img',
        'allow_coordinates_saving': True,
        'always_save_coordinates': True
    }
}
```

We have also never used the custom-fields "coordinates" - so they do not exist. Netbox runs on a fresh installed Alma Linux 9. With NB-Version 3.7.8 and Plugin-Version 3.9.1 everything works fine.

We couldn't find any errors in the log-files. Do you have any ideas how to debug this error?

Thanks,
Phil

**Comment by dreng** [comment]
A custom field is not needed anymore. It fact, it's deprecated. allow_coordinates_saving should be sufficient. Please remove always_save_coordinates, restart NetBox or your server and try again. Make sure, you activate "Save Coordinates" in the filter pane. 

Can you provide the URL including all parameters (the part after your IP/domain is sufficient)?

**Comment by phlpr** [comment]
Thanks for your response - here are my URLs with the different settings:

always_save_coordinates = True: When its activated in the config, i cant set it in the Filter-Options, its always YES but not set in the URL:
plugins/netbox_topology_views/topology/?show_unconnected=True&id=13746
Behavior: Floating

always_save_coordinates = False, "Save Coordinates" set in "Individual Options", i cant set it in the Filter-Options, its always YES but not set in the URL:
plugins/netbox_topology_views/topology/?show_unconnected=True&id=13746
Behavior: Floating

always_save_coordinates = False, "Save Coordinates" set in "Filter Options", now the URL changed:
plugins/netbox_topology_views/topology/?save_coords=True&show_unconnected=True&id=13746
Behavior: Still Floating

I also added a short video.

Thanks,
Phil

https://github.com/user-attachments/assets/d5af0c3f-71cc-4d82-82bf-c3f0403ae6ef

**Comment by dreng** [comment]
I am not able to reproduce this. I just tried with NetBox 4.10 and NTV 4.0.1 and different combinations of parameters and config options.

If you stop the NetBox service and start NetBox in foreground (manage.py runserver, see [installation manual),](https://netboxlabs.com/docs/netbox/en/stable/installation/3-netbox/) you should see something like

> "PATCH /api/plugins/netbox_topology_views/save-coords/save_coords/ HTTP/1.1" 200 25

after moving the icon. It's called in order to write the coordinates to the database. 200 is the HTTP ok status.

Beyond that, I'm at a bit of a loss as to how we can approach the problem without a development environment.

**Comment by phlpr** [comment]
Hi again,

i just tested it with mange.py runserver and watched the logs. I never get a "PATCH" Command, sometimes if i move the device long enough i can find a position where it sticks. But also then i get no "PATCH". The only log entry after i click the search-button:
![image](https://github.com/user-attachments/assets/81da03b9-a130-4e46-8d34-1454e13cc776)

Are there any 3rd party changes in NTV 4.0.0? Else i try to integrate it in my development environment and try do debug it.

P.S.: Before i debug it, i will test it with an empty database.

Thanks,
Philipp

**Reply by phlpr** [reply]
The problem still exists with an empty database

**Reply by mattieserver** [reply]
Never mind, the data would not show if the javascript was not loaded.

**Reply by mattieserver** [reply]
Can you check the 'source code' of the page and look for id_save_coords.

It should look something like this: 

![image](https://github.com/user-attachments/assets/ff63a467-5df7-4008-8c9d-52c276ce33f5)

**Reply by phlpr** [reply]
I found this part in the page source:
![image](https://github.com/user-attachments/assets/e90b4079-f8be-4d94-bee2-62738bde056c)

I also checked what happens when i click on "search":
NB 4.0.10
![image](https://github.com/user-attachments/assets/6913c3fb-f5f2-40d4-8b82-ad319b7fa2a5)

NB 3.7.8
![image](https://github.com/user-attachments/assets/0e203923-cc05-4619-8959-a80ed5c57925)

When I move a Device in NB 3.7.8 i see the following entry - in NB 4.0.10 nothing happens:
![image](https://github.com/user-attachments/assets/7c46aaa3-3147-4d6a-9225-505b7fa044b0)

In our setup i install all plugin's "offline". I just uninstalled NTV Plugin and deleted the complete netbox/netbox/static folder, reinstalled via pip (online) and ran "collectstatic" to ensure that i test with the correct version.

**Comment by phlpr** [comment, marked as answer]
You pointed me in the right direction.

The issue was that in my setup, I had overridden the original *app.js* and was still using the old version (from NTV 3.9.1). When I updated to the new version, I forgot to reset the cache.

Thanks to everyone for the help!

Best regards,  
Phil
